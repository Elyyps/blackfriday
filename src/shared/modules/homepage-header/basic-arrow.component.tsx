import * as React from "react";
import { IconComponent } from "@app/prep/modules-prep/core";
import { getArrow } from "@app/constants/icons";
import styles from "./homepage-header-component.module.scss";

interface IBasicArrowProps {
  className?: string;
  customArrow?: JSX.Element;
  onClick?: () => void;
  prev?: boolean;
}

const BasicArrow = (props: IBasicArrowProps) => {
  const { className, onClick } = props;
  let result;
  const customStyle = "homepage-header__slick-arrow-" + `${props.prev ? "prev" : "next"}`;
  if (props.customArrow === undefined) {
    result = (
      <button
        className={`${className} ${styles["homepage-header__slick-arrow"]} ${styles[customStyle]}`}
        style={{ display: "block" }}
        onClick={onClick}
      >
        <IconComponent icon={getArrow(props.prev)} size={"7px"} />
      </button>
    );
  } else {
    result = (
      <div className={className} onClick={onClick} role="button" aria-label="nextButton">
        {props.customArrow}
      </div>
    );
  }

  return result;
};
export { BasicArrow };
