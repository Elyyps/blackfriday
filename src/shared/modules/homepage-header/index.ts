export * from "./components/homepage-header.component";
