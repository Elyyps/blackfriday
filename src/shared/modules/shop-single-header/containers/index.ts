export * from "./shop-single-header-container";
