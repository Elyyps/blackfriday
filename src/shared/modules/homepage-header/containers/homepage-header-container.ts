import { connect, MapStateToProps } from "react-redux";

import { IAppState } from "@app/stores";
import { IBlackFridayDate } from "@app/stores/settings";
import { HomePageHeaderComponent, IHomePageHeaderComponentProps } from "../components/homepage-header.component";

interface IStateProps {
  blackFridayDate: IBlackFridayDate | undefined;
  totalResults: number;
}

const mapStateToProps: MapStateToProps<IStateProps, IHomePageHeaderComponentProps, IAppState> = state => {
  const blackFridayDate = state.settings.blackFridayDate;
  const { totalResults } = state.storeOverview;

  return { blackFridayDate, totalResults };
};

export type HomePageHeaderContainerProps = IStateProps;
export const HomePageHeaderContainer = connect(mapStateToProps)(HomePageHeaderComponent);
