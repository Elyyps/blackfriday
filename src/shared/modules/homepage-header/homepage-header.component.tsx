import * as React from "react";
import styles from "./homepage-header-component.module.scss";

import Slider, { Settings } from "react-slick";

import { CHEVRON_RIGHT } from "@app/constants/icons";
import { ArrowPanel, LabelComponent } from "@app/core";
import { getHomePageHeaderSetting } from "./slider-settings";
import { LinkComponent } from "@app/core/link";
import { HomepageHeader } from "@app/api/modules/homepage-header/homepage-header.module";

export interface IHomePageHeaderComponentProps {
  customNextArrow?: JSX.Element | undefined;
  customPrevArrow?: JSX.Element | undefined;
  homePageModule: HomepageHeader;
}

const HomePageHeaderComponent = (props: IHomePageHeaderComponentProps) => {
  const settings: Settings = getHomePageHeaderSetting(
    1,
    props.customNextArrow,
    props.customPrevArrow,
    styles["homepage-header__slick-slider"]
  );

  return (
    <div className={styles["homepage-header"]}>
      <div className={styles["homepage-header-top"]}>
        <div className="uk-container">
          <div className="uk-grid uk-child-width-1-2@m">
            <div>
              <div className={styles["homepage-header-wrap"]}>
                <LabelComponent text={"Black Friday 2019: NOG 10 dagen!"} />
                <div className={styles["homepage-header__title"]}>
                  <h1>
                    <strong>{props.homePageModule.title}</strong>
                  </h1>
                </div>
                <ul className={`uk-list-check ${styles["homepage-header__list"]}`}>
                  {props.homePageModule.description.map((text, index) => (
                    <li key={index}>{text}</li>
                  ))}
                </ul>
                <div className={styles["homepage-header__cards"]}>
                  {props.homePageModule.arrowPanelslinks.map((panel, key) => (
                    <ArrowPanel key={key} title={panel.text.title} icon={panel.icon} link={panel.text.url} />
                  ))}
                </div>
              </div>
              <div className={`uk-invert ${styles["homepage-header-info"]}`}>
                <div className={styles["homepage-header-info__col"]}>
                  <div className={styles["homepage-header-info__label"]}>{props.homePageModule.startingDateText}</div>
                  <div className={styles["homepage-header-info__item"]}>
                    <h3>{props.homePageModule.startingDate}</h3>
                  </div>
                </div>
                <div className={styles["homepage-header-info__col"]}>
                  <div className={styles["homepage-header-info__label"]}>
                    {props.homePageModule.durationBlackFridayText}
                  </div>
                  <div className={styles["homepage-header-info__item"]}>
                    <h3>{props.homePageModule.durationBlackFriday}</h3>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <Slider {...settings}>
                {props.homePageModule.slides.map((slide, index) => (
                  <div key={index} className={styles["homepage-header__slide-container"]}>
                    <div
                      style={{ backgroundImage: `url(${slide.image})` }}
                      className={styles["homepage-header__slider-item"]}
                    />
                    <div className={styles["homepage-header__slider-control"]}>
                      <div className={styles["homepage-header__slider-title"]}> {slide.title}</div>
                      <LinkComponent icon={CHEVRON_RIGHT} to={slide.link.url} animated variant={"quaternary"}>
                        {slide.link.title}
                      </LinkComponent>
                    </div>
                  </div>
                ))}
              </Slider>
            </div>
          </div>
        </div>
      </div>
      <div className={styles["homepage-header-bottom"]}>
        <div className={`uk-invert ${styles["homepage-header-bottom__wrap"]}`}>
          <div className="uk-container">
            <div className="uk-grid uk-child-width-1-2@s">
              <div>
                <div className={styles["homepage-header-bottom__label"]}>
                  <h2>{props.homePageModule.popularProductsTitle}</h2>
                </div>
                <div className={styles["homepage-header-bottom__cards"]}>
                  {props.homePageModule.popularProducts.map((product, key) => (
                    <div key={key}>
                      <ArrowPanel title={product.text.title} icon={product.icon} link={product.text.url} />
                    </div>
                  ))}
                </div>
              </div>
              <div>
                <div className={styles["homepage-header-bottom__label"]}>
                  <h2>{props.homePageModule.popularStoresTitle}</h2>
                </div>
                <div className={styles["homepage-header-bottom__cards"]}>
                  {props.homePageModule.popularStores.map((store, key) => (
                    <div key={key}>
                      <ArrowPanel title={store.text.title} icon={store.icon} link={store.text.url} />
                    </div>
                  ))}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export { HomePageHeaderComponent };
