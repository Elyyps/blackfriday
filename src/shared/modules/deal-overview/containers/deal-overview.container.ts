import { connect, MapStateToProps, MapDispatchToProps } from "react-redux";
import { IDealOverviewComponentProps, DealOverview } from "@app/modules/deal-overview";
import { IAppState } from "@app/stores";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { dealOverviewActions } from "@app/stores/deal-overview";
import { Deal } from "@app/api/core/deal/deal";
import { dealOverviewThunks } from "@app/middleware/thunk/deal-overview.thunk";
import { IScreenSize } from "@app/stores/settings";

interface IStateProps {
  brandFilterItems: FilterItem[];
  categoryFilterItems: FilterItem[];
  deals: Deal[];
  screenSize: IScreenSize | undefined;
  sortBy: string;
  storeFilterItems: FilterItem[];
  totalResults: number;
}

interface IDispatchProps {
  clearAllFilters: () => void;
  getDeals: (
    skip: number,
    take: number,
    storeFilters: FilterItem[],
    categoryFilters: FilterItem[],
    brandFilters: FilterItem[],
    sortBy?: string
  ) => void;
  setBrandFilters: (filterItems: FilterItem[]) => void;
  setCategoryFilters: (filterItems: FilterItem[]) => void;
  setSortBy: (sortBy: string) => void;
  setStoreFilters: (filterItems: FilterItem[]) => void;
}

const mapStateToProps: MapStateToProps<IStateProps, IDealOverviewComponentProps, IAppState> = state => {
  const { brandFilterItems, categoryFilterItems, sortBy, storeFilterItems, deals, totalResults } = state.dealOverview;

  const { screenSize } = state.settings;

  return {
    brandFilterItems,
    categoryFilterItems,
    screenSize,
    sortBy,
    storeFilterItems,
    deals,
    totalResults
  };
};

const mapDispatchToProps: MapDispatchToProps<IDispatchProps, IDealOverviewComponentProps> = dispatch => ({
  getDeals: (
    skip: number,
    take: number,
    storeFilters: FilterItem[],
    categoryFilters: FilterItem[],
    brandFilters: FilterItem[],
    sortBy?: string
  ) => dispatch(dealOverviewThunks.getStores(skip, take, storeFilters, categoryFilters, brandFilters, sortBy)),
  setBrandFilters: (filterItems: FilterItem[]) => dispatch(dealOverviewActions.setBrandFilters({ filterItems })),
  setCategoryFilters: (filterItems: FilterItem[]) => dispatch(dealOverviewActions.setCategoryFilters({ filterItems })),
  setSortBy: (sortBy: string) => dispatch(dealOverviewActions.setSortBy({ sortBy })),
  setStoreFilters: (filterItems: FilterItem[]) => dispatch(dealOverviewActions.setStoreFilters({ filterItems })),
  clearAllFilters: () => dispatch(dealOverviewActions.clearAllFilters())
});

export type DealOverviewContainerProps = IStateProps & IDispatchProps;
export const DealOverviewContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(DealOverview);
