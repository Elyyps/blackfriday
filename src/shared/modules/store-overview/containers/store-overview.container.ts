import { connect, MapStateToProps, MapDispatchToProps } from "react-redux";
import { IStoreOverviewComponentProps, StoreOverview } from "@app/modules/store-overview";
import { IAppState } from "@app/stores";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { storeOverviewActions } from "@app/stores/store-overview";
import { Store } from "@app/api/core/store/store";

interface IStateProps {
  brandFilterItems: FilterItem[];
  categoryFilterItems: FilterItem[];
  sortBy: string;
  statusFilterItems: FilterItem[];
  stores: Store[];
}
interface IDispatchProps {
  setBrandFilters: (filterItems: FilterItem[]) => void;
  setCategoryFilters: (filterItems: FilterItem[]) => void;
  setSortBy: (sortBy: string) => void;
  setStatusFilters: (filterItems: FilterItem[]) => void;
  setStores: (stores: Store[]) => void;
}

const mapStateToProps: MapStateToProps<IStateProps, IStoreOverviewComponentProps, IAppState> = state => {
  const { brandFilterItems, categoryFilterItems, sortBy, statusFilterItems, stores } = state.storeOverview;

  return {
    brandFilterItems,
    categoryFilterItems,
    sortBy,
    statusFilterItems,
    stores
  };
};

const mapDispatchToProps: MapDispatchToProps<IDispatchProps, IStoreOverviewComponentProps> = dispatch => ({
  setBrandFilters: (filterItems: FilterItem[]) => dispatch(storeOverviewActions.setBrandFilters({ filterItems })),
  setCategoryFilters: (filterItems: FilterItem[]) => dispatch(storeOverviewActions.setCategoryFilters({ filterItems })),
  setSortBy: (sortBy: string) => dispatch(storeOverviewActions.setSortBy({ sortBy })),
  setStatusFilters: (filterItems: FilterItem[]) => dispatch(storeOverviewActions.setStatusFilters({ filterItems })),
  setStores: (stores: Store[]) => dispatch(storeOverviewActions.setStores({ stores }))
});

export type StoreOverviewContainerProps = IStateProps & IDispatchProps;
export const StoreOverviewContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(StoreOverview);
