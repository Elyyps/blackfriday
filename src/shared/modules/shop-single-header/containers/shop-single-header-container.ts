import { connect, MapStateToProps } from "react-redux";
import { ShopSingleHeaderComponent, IShopSingleHeaderComponentProps } from "../shop-single-header.component";
import { IAppState } from "@app/stores";
import { IScreenSize } from "@app/stores/settings";

interface IStateProps {
  screenSize: IScreenSize | undefined;
}

const mapStateToProps: MapStateToProps<IStateProps, IShopSingleHeaderComponentProps, IAppState> = state => {
  const { screenSize } = state.settings;

  return {
    screenSize
  };
};

export type ShopSingleHeaderContainerProps = IStateProps;
export const ShopSingleHeaderContainer = connect(mapStateToProps)(ShopSingleHeaderComponent);
