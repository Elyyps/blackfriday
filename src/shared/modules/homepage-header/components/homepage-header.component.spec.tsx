import React from "react";
import { shallow } from "enzyme";

import { getHeaderpageHeader } from "@app/api/modules/homepage-header/endpoints";
import { HomePageHeaderContainer } from "../containers/homepage-header-container";
describe("[homepageHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HomePageHeaderContainer homePageModule={getHeaderpageHeader()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
