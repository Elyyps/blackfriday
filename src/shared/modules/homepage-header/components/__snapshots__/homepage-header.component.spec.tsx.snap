// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[homepageHeader] should render component without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
